### PR TITLE
feat(medusa): Prevent cart completion conflict

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
Remove serializable transactions and improve perf. Prevent unnecessary conflicting requests. Goes along the pr for stripe delay and retrial from @olivermrbl

### CODE REVIEW

**Comment by adrien2p** [comment]
@olivermrbl this one needs validation with stripe to test the entire flow

File: packages/medusa/src/services/cart.ts [hunk 1]
```diff
@@ -1899,31 +1905,32 @@ class CartService extends TransactionBaseService {
...
-              "region",
-              "region.tax_rates",
-              "region.payment_providers",
-              "payment_sessions",
-              "customer",
-            ],
-          },
-          { force_taxes: true }
-        )
+        const cartId = isString(cartOrCartId) ? cartOrCartId : cartOrCartId.id
```
Comment (olivermrbl) [inline packages/medusa/src/services/cart.ts]: comment: Seems unnecessary with the second ternary in place. You should be able to replace `cartId` with `cartOrCartId` in the `retrieveWithTotals` call and remove this

**Review by olivermrbl — APPROVED** [review]
can I get you to add a description to the PR, we can use for the release notes?

**Comment by adrien2p** [comment]
done @olivermrbl

**Comment by olivermrbl** [comment]
Looks like tests will need updating :)

**Comment by adrien2p** [comment]
> Looks like tests will need updating :)

on it 💪

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**